src/models/MyAccount.ts

```diff
@@ -0,0 +1,213 @@
+import { sierraClient } from "../server/sierraClient"
+import type {
+  Checkout,
+  Hold,
+  Patron,
+  Fine,
+  SierraAccountData,
+  SierraCheckout,
+  SierraHold,
+  SierraCodeName,
+  SierraPatron,
+  SierraFine,
+  SierraFineEntry,
+} from "../types/accountTypes"
+
+let client
+
+export default class MyAccount {
+  checkouts: Checkout[]
+  holds: Hold[]
+  patron: Patron
+  fines: Fine
+  constructor({
+    checkouts,
+    holds,
+    patron,
+    fines,
+    checkoutBibData,
+    holdBibData,
+  }: SierraAccountData) {
+    this.checkouts = this.buildCheckouts(checkouts, checkoutBibData)
+    this.holds = this.buildHolds(holds, holdBibData)
+    this.patron = this.buildPatron(patron)
+    this.fines = this.buildFines(fines)
+  }
+
+  static async MyAccountFactory(id) {
+    client = await sierraClient()
+    const baseQuery = `patrons/${id}`
+    const checkouts = await this.fetchCheckouts(baseQuery)
+    const holds = await this.fetchHolds(baseQuery)
+    const patron = await this.fetchPatron(baseQuery)
+    const fines = await this.fetchFines(baseQuery)
+    const checkoutBibData = await this.fetchBibData(checkouts.entries, "item")
+    const holdBibData = await this.fetchBibData(holds.entries, "record")
+    return new this({
+      checkouts: checkouts.entries,
+      holds: holds.entries,
+      patron,
+      fines,
+      checkoutBibData,
+      holdBibData,
+    })
+  }
+
+  buildCheckouts(checkouts: SierraCheckout[], bibDataMap): Checkout[] {
+    return checkouts.map((checkout: SierraCheckout) => {
+      return {
+        id: MyAccount.getRecordId(checkout.id),
+        callNumber: checkout.item.callNumber,
+        barcode: checkout.item.barcode,
+        dueDate: checkout.dueDate,
+        patron: MyAccount.getRecordId(checkout.patron),
+        title: bibDataMap[checkout.item.bibIds[0]].title,
+        isResearch: bibDataMap[checkout.item.bibIds[0]].isResearch,
+        bibId: checkout.item.bibIds[0],
+        isNyplOwned: bibDataMap[checkout.item.bibIds[0]].isNyplOwned,
+      }
+    })
+  }
+
+  static async fetchCheckouts(baseQuery: string) {
```

Comment:
I better understand this patter now. The two `this`s are confusing in the factory function and the constructor but it makes sense so you can ignore this comment.

It would be nice to somehow have the patron id as an instance variable so that here you could do
```
client.get(`${this.patronId}${holdsQuery}`)
```

and not have to pass `baseQuery`, but this function gets called before the instance is created. That also confused me a bit.